src/platforms/esp32/main/socket_driver.c

```diff
@@ -471,12 +471,13 @@ term socket_driver_do_send(Context *ctx, term buffer)
         buf = term_binary_data(buffer);
         len = term_binary_size(buffer);
     } else if (term_is_list(buffer)) {
+        int proper;
+        len = term_list_length(buffer, &proper);
         int ok;
         buf = interop_list_to_string(buffer, &ok);
-        if (!ok) {
+        if (UNLIKELY(!proper || !ok)) {
             // TODO error
         }
-        len = strlen(buf);
```

Comment:
Ouch, what was I thinking?  Presumably this would have crashed and burned with UTF-8, for example.